File: backend/general/routes/articles.go
package routes

import (
	"general/types"
	"general/utils"
	"net/http"

	"github.com/gin-gonic/gin"
)

type ArticlesHandler struct {
	router Router
}

func (ah *ArticlesHandler) Run() (err error) {
	ah.router.Get("/articles", ah.get)
	return
}

func (ah *ArticlesHandler) Stop()

func (ah *ArticlesHandler) get(c *gin.Context) {
	page, size := getPageSize(c.Request)
	listType := getParam(c.Request, "type")
	tag := getParam(c.Request, "tag")
	userId := getUserIdFromQuery(c.Request)
	selfUserId := utils.IsAuth(c.Request)
	var articles []types.ArticleListData
	switch listType {
	case "view":
		articles = services.GetViewList(page, size, userId)
	case "hot":
		articles = services.GetHotList(page, size, userId)
	case "profile":
		articles = services.GetProfileList(page, size, userId, selfUserId)
	case "tag":
		articles = services.GetTagList(page, size, tag)
	default:
		articles = services.GetNewestList(page, size, userId)
	}

	c.JSON(http.StatusOK, gin.H{"status": "success", "list": articles})
}

File: frontend/api/articles.ts
import { GENERAL_IP, SSR_GENERAL_IP } from "@/constants/env";
import api from "./utils";

export const getArticles = async (userId = 0, page = 1, isSSR = false) => {
  return await api.get(
    `${
      isSSR ? SSR_GENERAL_IP : GENERAL_IP
    }/articles?page=${page}&userId=${userId}`
  );
};

export const getViewArticles = async (userId = 0, page = 1, isSSR = false) => {
  return await api.get(
    `${
      isSSR ? SSR_GENERAL_IP : GENERAL_IP
    }/articles?type=view&page=${page}&userId=${userId}`
  );
};

export const getHotArticles = async (userId = 0, page = 1, isSSR = false) => {
  return await api.get(
    `${
      isSSR ? SSR_GENERAL_IP : GENERAL_IP
    }/articles?type=hot&page=${page}&userId=${userId}`
  );
};

export const getProfileArticles = async (userId = 0, page = 1) => {
  return await api.get(
    `${GENERAL_IP}/articles?type=profile&page=${page}&userId=${userId}`
  );
};
